src/input/base-input.js

```diff
@@ -84,7 +85,7 @@ class BaseInput<T: EventTarget> extends React.Component<
     } = this.props;
 
     return {
-      $ref: inputRef,
+      $ref: $ref || inputRef,
```

Comment:
Thinking of deprecating `inputRef` and use `$ref` from the top level props since we don't do props spreading on input elements.